src/rules/no-multiple-h1.js

```diff
@@ -110,8 +110,10 @@ export default {
 			},
 
 			html(node) {
+				const text = stripHtmlComments(node.value);
+
 				let match;
-				while ((match = h1TagPattern.exec(node.value)) !== null) {
+				while ((match = h1TagPattern.exec(text)) !== null) {
```

Comment:
Now we are matching with regular expression twice, wouldn't this make the rule slower?